### PR TITLE
[NAE-1677] Message from exception thrown in SET event on data field is not propagated in EventOutcome

### PR DESCRIPTION
**Pull request by Coresto**
# Description

Refactoring of error handling so that error messages are displayed on the frontend

Fixes [NAE-1677]

## Dependencies

- No new dependencies were introduced

### Third party dependencies

- No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

I created net with events on set and get in witch an exception was thrown and after that message from exception showed in snack bar as error.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |  Windows 10         |
| Runtime             | Node 14.19.3        |
| Dependency Manager  |  NPM  6.14.17        |
| Framework version   | Angular 13          |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My changes have been checked, personally or remotely, with @machacjozef & @RudeBoyxX 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [ ] I have checked my contribution with code analysis tools:
    - [ ] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [ ] I have made corresponding changes to the documentation:
    - [ ] Developer documentation
    - [ ] User Guides
    - [ ] Migration Guides

[NAE-1677]: https://netgrif.atlassian.net/browse/NAE-1677?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.ts [hunk 1]
```diff
@@ -204,41 +204,45 @@ export class TaskResourceService extends AbstractResourceService implements Coun
...
-                const dataGroupsArray = this.changeType((responseOutcome.outcome as GetDataGroupsEventOutcome).data, 'dataGroups');
-                if (!Array.isArray(dataGroupsArray)) {
-                    return [];
-                }
-                const result = [];
-                dataGroupsArray.forEach(dataGroupResource => {
-                    const dataFields: Array<DataField<any>> = [];
-                    if (!dataGroupResource.fields._embedded) {
-                        return; // continue
+                if (responseOutcome.success) {
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.ts:207]: the condition should be inverted and only the `else` block should be a part of the if. The rest of the code then remains more readable as it has less indentation

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts [hunk 2]
```diff
@@ -239,7 +239,7 @@ export class TaskDataService extends TaskHandlingService implements OnDestroy {
      * @param nextEvent indicates to the event queue that the next event can be processed
      */
     protected processErroneousGetDataRequest(gottenTaskId: string,
-                                             error: HttpErrorResponse,
+                                             error: any,
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/task/services/task-data.service.ts:242]: why `any`? If the newly added type is `Error` then the type can be extended into a union `HttpErrorResponse | Error`

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts [hunk 3]
```diff
@@ -428,7 +430,7 @@ export class TaskDataService extends TaskHandlingService implements OnDestroy {
      * @param body hold the data that was sent in request
      */
     protected processSuccessfulSetDataRequest(setTaskId: string,
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/task/services/task-data.service.ts:432]: if the method is called `processSuccessfulSetDataRequest` it should only do that and it should not check if the response was indeed a success. That should be resolved elsewhere

**Review by minop — CHANGES_REQUESTED** [review]

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts (hunk 2 above)
Comment (Coresto) [reply projects/netgrif-components-core/src/lib/task/services/task-data.service.ts]: Done

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts (hunk 3 above)
Comment (Coresto) [reply projects/netgrif-components-core/src/lib/task/services/task-data.service.ts]: Done

File: projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.ts (hunk 1 above)
Comment (Coresto) [reply projects/netgrif-components-core/src/lib/resources/engine-endpoint/task-resource.service.ts]: Done

**Review by minop — APPROVED** [review]

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts [hunk 4]
```diff
@@ -413,41 +415,78 @@ export class TaskDataService extends TaskHandlingService implements OnDestroy {
...
+    protected processUnsuccessfulSetDataRequest(setTaskId: string,
+                                              response: EventOutcomeMessageResource,
                                               afterAction: AfterAction,
                                               nextEvent: AfterAction,
                                               body: TaskSetDataRequestBody) {
-        if (!this.isTaskRelevant(setTaskId)) {
-            this._log.debug('current task changed before the set data response could be received, discarding...');
-            this._taskState.stopLoading(setTaskId);
-            this._taskState.stopUpdating(setTaskId);
-            afterAction.complete();
```
Comment (Kovy95) [inline projects/netgrif-components-core/src/lib/task/services/task-data.service.ts:439]: it is okay that afterAction.complete and stop loading and updating was removed ?

Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/task/services/task-data.service.ts:439]: implement in performSetDataRequest

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts [hunk 5]
```diff
@@ -413,41 +415,78 @@ export class TaskDataService extends TaskHandlingService implements OnDestroy {
 
         this._taskResourceService.setData(this._safeTask.stringId, body).pipe(take(1))
             .subscribe((response: EventOutcomeMessageResource) => {
-                this.processSuccessfulSetDataRequest(setTaskId, response.outcome, afterAction, nextEvent, body);
+                if (!this.isTaskRelevant(setTaskId)) {
```
Comment (machacjozef) [inline projects/netgrif-components-core/src/lib/task/services/task-data.service.ts:418]: here 😺

**Review by mladoniczky — APPROVED** [review]

**Review by mladoniczky — DISMISSED** [review]

File: projects/netgrif-components-core/src/lib/task/services/task-data.service.ts (hunk 4 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/task/services/task-data.service.ts:439]: its okey...

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**